### PR TITLE
Add tests for changes to call devices.

### PR DESCRIPTION
**Pull request by richardfazzio**

### CODE REVIEW

File: test/browser/call-device.js [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+describe('Call Devices', async function() {
+    this.timeout(300000);
+
+    before(async function() {
+        Circuit.logger.setLevel(Circuit.Enums.LogLevel.Error);
+        client = new Circuit.Client(config.config);
+        const res = await Promise.all([PeerUser.create(), client.logon(config.credentials)]);
+        peerUser = res[0];
+        user = res[1];
+        const conversation = await client.createGroupConversation([peerUser.userId], 'SDK Test: Conference Call');
```
Comment (rogeru) [inline test/browser/call-device.js]: is "SDK Test: Conference Call" right?

File: test/browser/call-device.js [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+    });
+
+    it('should get media devices and confirm the correct client', async () => {
+        const res = await client.getDevices();
+        assert(res.some(dev => dev.clientId === user.clientId));
+    });
+
+    it('should get audio and video stats', async () => {
+        const res = await client.getAudioVideoStats();
+        assert(res);
```
Comment (rogeru) [inline test/browser/call-device.js]: Would be good to check a bit more, e.g. that there are packets being sent/received

File: test/browser/call-device.js [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+    });    
+    
+    it('should get local audio and video stream', async () => {
+        stream = await client.getLocalAudioVideoStream();
+        assert(stream);
+    });    
+    it('should set local audio and video stream', async () => {
+        await client.setAudioVideoStream(call.callId, stream);
```
Comment (rogeru) [inline test/browser/call-device.js]: In the future we should send an actual stream we create locally, e.g. using canvas or a static video, and then validate somehow the steam is received on the other side. But for now I guess just setting the same local stream is fine

File: test/browser/call-device.js [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+        const audioOutputDevice = mediaDevices.find(d => d.kind === 'audiooutput');
+        let devices = {
+            recording: audioInputDevice.deviceId,
+            playback: audioOutputDevice.deviceId,
+            video: videoInputDevice.deviceId,
+            ringiing: audioOutputDevice.deviceId
+        }
+        const r = await Promise.all([
+            client.setMediaDevices(devices),
+            expectEvents(client, [{
```
Comment (rogeru) [inline test/browser/call-device.js]: Here we aren't really testing anything regarding devices. But again that's difficult. For now that's fine.

File: test/browser/call-device.js (hunk 1 above)
Comment (richardfazzio) [reply test/browser/call-device.js]: No, i will make each conversation's title reflect the test.

File: test/browser/call-device.js (hunk 2 above)
Comment (richardfazzio) [reply test/browser/call-device.js]: I ended up checking that one element it returns is of type 'codec', I'm not sure if there would be something better to check.

**Pull request merged**